### DISCUSSION TITLE
Why is there a limit on plugins for the self-hosted version since plugins are community contributed?

### DISCUSSION THREAD

**Discussion by lolcabanon** [Budibase/budibase#10826] [General]
Hey! Just wanted to open a discussion on the subject of plugins limit of 10 for self-hosted free version.

**Since plugins comes mostly from the community contributions, wouldn't it be more logical to be able to install unlimited plugins on a self-hosted instance?**

I guess other limits makes more sense (env variables, audits, ...) since they add advanced functionalities, and they make the DX better and simpler, therefore it feels right that they are paying features. Also, they are built by the Budibase team and I respect the work behind it.

Curious to know what you have to say about that!

Edit : It has been discussed in #7874 but there seems to be no other answer than "it's enough"...

**Comment by mjashanks** [comment]
Hey @lolcabanon 

> Since plugins comes mostly from the community contributions, wouldn't it be more logical to be able to install unlimited plugins on a self-hosted instance?

The simple answer is that we have to choose where to put pay-gates, and this was one of them. 10 feels quite generous - very few users hit this limit.

Plugins are meant to fill a gap - components and data sources that we don't have in the core product. We're always willing to bring in new components to the core product, which would lower the barrier for you. 

Which plugins do you wish we would adopt into the core product?

**Reply by DeadGhost1407** [reply]
Hi @mjashanks,
I love budibase and the hard work you guys put into it. So if you say 10 is quite generous, I won’t disagree. 

But, since some plugins require 2-3 companion plugins, and since a bunch of functionality that i need is unfortunately missing from budibase, 10 is no where near enough. See below for my “brief” list of plugins and usage. 

Again, thank you to the bb team, but If I wish you guys would reconsider the plugin limit at least temporarily until the core components are a bit more robust (at least compared to some tools like Retool, tooljet, appsmith…etc). 

1- tooltip (not sure why this is not core since bb had tooltips and onboarding walkthrough…etc)
2- super table (the core table is missing a lot of options that these 3 plugin combo has)
3- super column
4- super cell
5- interval (i use it for session timeout, running code periodically to auto refresh some data… this is very important to my use cases)
6- accordion (for organization and decluttering the UI)
7- file upload (dealing up attachments is a pain with the core functionality. Still a pain with this plugin but much much better imo)
8- file repeater (needed for the file upload)
9- side navigation (the core navbar is very limited, can’t nest links for example, and unfortunately I have a few too many screen since I need to use state across an app)
10- csv import (since there is no easy way to do this from the front-end that I am aware of)

** i had to remove the icon status cards and rebuilt it by using a bunch of containers, icons, paragraphs…etc, which is not ideal.

i also removed the qr code scanner and but actual bluetooth scanners so do inventory. 

I spun up a different budibase instance since I needed the comment box plugin functionality, but I was able to figure out an alternative/workaround

the kanban board, calendar view, qr code, and signature were all removed (even though we had a use case for them - i work in a small retail shop). 

sorry about the long post. Have a nice day!
